tests/db/test_bigchain_api.py

```diff
@@ -232,204 +99,22 @@ def test_text_search(self, b, alice):
         else:
             assert len(assets) == 3
 
-    @pytest.mark.genesis
-    def test_text_search_returns_valid_only(self, monkeypatch, b, alice):
-        from bigchaindb.models import Transaction
-        from bigchaindb.backend.exceptions import OperationError
-        from bigchaindb.backend.localmongodb.connection import LocalMongoDBConnection
-
-        asset_valid = {'msg': 'Hello BigchainDB!'}
-        asset_invalid = {'msg': 'Goodbye BigchainDB!'}
-
-        monkeypatch.setattr('time.time', lambda: 1000000000)
-        tx1 = Transaction.create([alice.public_key], [([alice.public_key], 1)],
-                                 asset=asset_valid)
-        tx1 = tx1.sign([alice.private_key])
-        block1 = b.create_block([tx1])
-        b.write_block(block1)
-
-        monkeypatch.setattr('time.time', lambda: 1000000020)
-        tx2 = Transaction.create([alice.public_key], [([alice.public_key], 1)],
-                                 asset=asset_invalid)
-        tx2 = tx2.sign([alice.private_key])
-        block2 = b.create_block([tx2])
-        b.write_block(block2)
-
-        # vote the first block valid
-        vote = b.vote(block1.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
-
-        # vote the second block invalid
-        vote = b.vote(block2.id, b.get_last_voted_block().id, False)
-        b.write_vote(vote)
-
-        # get assets with text search
-        try:
-            assets = list(b.text_search('bigchaindb'))
-        except OperationError:
-            assert not isinstance(b.connection, LocalMongoDBConnection)
-            return
-
-        # should only return one asset
-        assert len(assets) == 1
-        # should return the asset created by tx1
-        assert assets[0] == {
-            'data': {'msg': 'Hello BigchainDB!'},
-            'id': tx1.id
-        }
-
-    @pytest.mark.usefixtures('inputs')
-    def test_write_transaction(self, b, user_pk, user_sk):
-        from bigchaindb.models import Transaction
-
-        input_tx = b.get_owned_ids(user_pk).pop()
-        input_tx = b.get_transaction(input_tx.txid)
-        inputs = input_tx.to_inputs()
-        tx = Transaction.transfer(inputs, [([user_pk], 1)],
-                                  asset_id=input_tx.id)
-        tx = tx.sign([user_sk])
-        b.write_transaction(tx)
-
-        tx_from_db, status = b.get_transaction(tx.id, include_status=True)
-
-        assert tx_from_db.to_dict() == tx.to_dict()
-
-    @pytest.mark.usefixtures('inputs')
-    def test_read_transaction(self, b, user_pk, user_sk):
```

Comment:
same here, why is it removed?

Reply:
I replaced the first test. This one seems very similar to the first... Basically, we write something to the DB, get it back, and check that nothing has changed. I don't see a way to keep these tests distinct with BDB2.0

Reply:
I see. So where is the first test? You said you replaced it?